sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/TabCharacterCheck.java

```diff
@@ -20,46 +20,61 @@
 package org.sonar.plugins.xml.checks;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.sonar.check.Rule;
-import org.sonar.check.RuleProperty;
 import org.sonar.plugins.xml.Utils;
+import org.sonar.plugins.xml.Xml;
+import org.sonarsource.analyzer.commons.annotations.DeprecatedRuleKey;
 import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.sonarsource.analyzer.commons.xml.XmlTextRange;
 import org.sonarsource.analyzer.commons.xml.checks.SonarXmlCheck;
 
 /**
  * RSPEC-105
  */
 @Rule(key = TabCharacterCheck.RULE_KEY)
+@DeprecatedRuleKey(ruleKey = "IllegalTabCheck", repositoryKey = Xml.REPOSITORY_KEY)
 public class TabCharacterCheck extends SonarXmlCheck {
 
-  public static final String RULE_KEY = "IllegalTabCheck";
-
-  @RuleProperty(key = "markAll", description = "Mark all tab errors", defaultValue = "false", type = "BOOLEAN")
-  private boolean markAll;
+  public static final String RULE_KEY = "S105";
+  private static final Pattern TABS_REGEX = Pattern.compile("\t++");
+  private static final int MAX_REPORTED_LOCATION = 21;
 
   @Override
   public void scanFile(XmlFile file) {
-    int lineNumber = 1;
-    List<Integer> secondaries = new ArrayList<>();
-
-    for (String line : Utils.splitLines(file.getContents())) {
-      if (line.indexOf('\t') != -1) {
-        secondaries.add(lineNumber);
-        if (!markAll) {
-          break;
+    if (file.getContents().indexOf('\t') == -1) {
```

Comment:
In order to avoid calling `file.getContents()` two times, I would store it in a variable (2nd time line 50).